goal_src/engine/game/settings.gc

```diff
@@ -5,5 +5,734 @@
+        )
+       )
+      ((= v1-1 'vibration)
+       (set!
+        (-> obj vibration)
+        (the-as symbol (-> (the-as connection s3-0) param1))
+        )
+       )
+      ((= v1-1 'auto-save)
+       (set!
+        (-> obj auto-save)
+        (the-as symbol (-> (the-as connection s3-0) param1))
+        )
+       )
+      ((= v1-1 'allow-pause)
+       (set!
+        (-> obj allow-pause)
+        (the-as symbol (-> (the-as connection s3-0) param1))
+        )
+       )
+      ((= v1-1 'allow-progress)
+       (set!
+        (-> obj allow-progress)
+        (the-as symbol (-> (the-as connection s3-0) param1))
+        )
+       )
+      ((= v1-1 'play-hints)
+       (set!
+        (-> obj play-hints)
+        (the-as symbol (-> (the-as connection s3-0) param1))
+        )
+       )
+      ((= v1-1 'movie)
+       (set!
+        (-> obj movie)
+        (the-as (pointer progress) (-> (the-as connection s3-0) param1))
+        )
+       )
+      ((= v1-1 'talking)
+       (set!
+        (-> obj talking)
+        (the-as (pointer progress) (-> (the-as connection s3-0) param1))
+        )
+       )
+      ((= v1-1 'spooling)
+       (set!
+        (-> obj spooling)
+        (the-as (pointer progress) (-> (the-as connection s3-0) param1))
+        )
+       )
+      ((= v1-1 'hint)
+       (set!
+        (-> obj hint)
+        (the-as (pointer process) (-> (the-as connection s3-0) param1))
+        )
+       )
+      ((= v1-1 'ambient)
+       (set!
+        (-> obj ambient)
+        (the-as (pointer progress) (-> (the-as connection s3-0) param1))
+        )
+       )
+      ((= v1-1 'common-page)
+       (let ((v1-89 (-> (the-as connection s3-0) param1)))
+        (cond
+         ((= (the-as symbol v1-89) 'set)
+          (set!
+           (-> obj common-page)
+           (logior
+            (-> obj common-page)
+            (the-as int (the-as int32 (-> (the-as connection s3-0) param3)))
+            )
+           )
+          )
+         ((= v1-89 'clear)
+          (set!
+           (-> obj common-page)
+           (logand
+            (-> obj common-page)
+            (the-as
+             int
+             (the-as
+              uint
+              (lognot (the-as uint (-> (the-as connection s3-0) param3)))
+              )
+             )
+            )
+           )
+          )
+         )
+        )
+       )
+      )
+     )
+    (set! s3-0 s4-0)
+    (set! s4-0 (-> s3-0 prev0))
+    )
+   )
+  obj
+  )
+
+;; definition for method 9 of type setting-control
+;; INFO: Return type mismatch int vs none.
+(defmethod
+  TODO-RENAME-9
+  setting-control
+  ((obj setting-control)
+   (arg0 process)
+   (arg1 (function object object object object object))
+   (arg2 object)
+   (arg3 object)
+   (arg4 object)
+   )
+  (add-connection (-> obj engine) arg0 arg1 arg2 arg3 arg4)
+  0
+  (none)
+  )
+
+;; definition for method 10 of type setting-control
+;; INFO: Return type mismatch int vs none.
+(defmethod
+  TODO-RENAME-10
```

Comment:
possible name: `set-setting!` - this will clear any pending settings from the same process first